towhee/runtime/data_queue.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Tuple
+import threading
+from enum import Enum, auto
+
+from collections import deque, namedtuple
+
+
+class DataQueue:
```

Comment:
why can't we directly wrap one of the Queue types in python queue module?

Reply:
Queue is a thread-safe wrapper of deque